Makefile

```diff
@@ -18,10 +18,32 @@ ifeq ($(DBG),1)
 GOGCFLAGS ?= -gcflags=all="-N -l"
 endif
 
-TOOLS_DIR=./tools
+#
+# Directories.
+#
+CLUSTER_API_DIR := third_party/cluster-api
+EXP_DIR := ${CLUSTER_API_DIR}/exp
+TOOLS_DIR=$(abspath ./tools)
 BIN_DIR=bin
 TOOLS_BIN_DIR := $(TOOLS_DIR)/$(BIN_DIR)
-CONTROLLER_GEN := $(abspath $(TOOLS_BIN_DIR)/controller-gen)
+GO_INSTALL := ./scripts/go_install.sh
+
+# Helper function to get dependency version from go.mod
+get_go_version = $(shell go list -m $1 | awk '{print $$2}')
+
+#
+# Binaries.
+#
+# Note: Need to use abspath so we can invoke these from subdirectories
+CONTROLLER_GEN_VER := v0.12.0
```

Comment:
Are we generating the CRDs from source? Any reason not to copy from upstream files?

Do we have a check to make sure that if these are out of date, the PR gets a red status check?

Reply:
We do have a script that syncs the CRDs.  I can update workflow to fail if CRDs are not as expected.